hbase-common/src/main/java/org/apache/hadoop/hbase/util/ClassSize.java

```diff
@@ -186,13 +186,19 @@ long sizeOfByteArray(int len) {
       private byte a;
     }
 
+    private static final int ARRAY_OBJECT_INDEX_SCALE =
+      HBasePlatformDependent.arrayIndexScale(Object[].class);
+
+    private static final int ARRAY_BYTE_INDEX_SCALE =
+      HBasePlatformDependent.arrayIndexScale(byte[].class);
+
     public UnsafeLayout() {
     }
 
     @Override
     int headerSize() {
       try {
-        return (int) UnsafeAccess.theUnsafe.objectFieldOffset(
+        return (int) HBasePlatformDependent.objectFieldOffset(
```

Comment:
This is platform dependent so should be moved into HBasePlatformDependent. Perhaps all of ClassSize should go up in there but at least these methods involving object layout concerns can be replaced with external static helpers in the thirdparty module.